code/extensions/che-remote/src/extension.ts

```diff
@@ -89,41 +97,96 @@ export function deactivate(): void {
 
 }
 
-async function updateDevfile(cheApi: any): Promise<void> {
+async function isFile(filePath: string): Promise<boolean> {
+  try {
+    if (await fs.pathExists(filePath)) {
+      const stat = await vscode.workspace.fs.stat(vscode.Uri.file(filePath));
+      return stat.type === vscode.FileType.File;
+    }
+  } catch (error) {
+    console.error(error);
+  }
+
+  return false;
+}
+
+async function selectDevfile(): Promise<string | undefined> {
+  const devfileItems: vscode.QuickPickItem[] = [];
+
+  const projects = await fs.readdir(process.env.PROJECTS_ROOT as string);
+  for (const project of projects) {
+    for (const devfileName of DEVFILE_NAMES) {
+      const devfilePath = path.join(process.env.PROJECTS_ROOT!, project, devfileName);
+      if (await isFile(devfilePath)) {
+        devfileItems.push({
+          label: devfilePath,
+          detail: project
+        });
+        break;
+      }
+    }
+  }
+
+  devfileItems.push({
+    kind: vscode.QuickPickItemKind.Separator,
+    label: 'alternative'
+  });
+
+  devfileItems.push({
+    label: `${process.env.PROJECTS_ROOT!}/*`,
+    detail: 'Select a Devfile with a different name'
+  });
+
+  const devfileItem = await vscode.window.showQuickPick(devfileItems, {
+    title: 'Select a Devfile to be applied to the current workspace',
```

Comment:
I have a workspace with only one project, the system proposes me to select **any file** (not devfile) for restarting the workspace:
<img width="1228" alt="Screenshot 2024-05-14 at 09 48 33" src="https://github.com/che-incubator/che-code/assets/5676062/0ea55b72-0194-425e-9a56-8e7e046564ab">

<img width="515" alt="Screenshot 2024-05-14 at 09 49 45" src="https://github.com/che-incubator/che-code/assets/5676062/5962ecc6-b488-461f-8995-4fcf87f5c3f9">

Do we really need it?

Reply:
It could be helpful in case you have a workspace with only one project and with a test devfile that has a different name.
User is responsible what does he do. If he select a binary file or something else it is expected the workspace will not be restarted.